config/initializers/content_security_policy.rb

```diff
@@ -35,6 +35,9 @@ def host_to_url(str)
 
     p.connect_src :self, :data, :blob, assets_host, media_host, Rails.configuration.x.streaming_api_base_url, *webpacker_urls
     p.script_src  :self, :unsafe_inline, :unsafe_eval, assets_host
+  elsif Rails.env.test?
+    p.connect_src :self, :data, :blob, assets_host, media_host, Rails.configuration.x.streaming_api_base_url
+    p.script_src  :self, assets_host, :unsafe_eval, "'wasm-unsafe-eval'"
```

Comment:
Why is `unsafe-eval` required here?

Reply:
I need to re-check, but IIRC this is needed by Selenium as it injects JS into the page to drive the tests

Reply:
This is caused by Webpack being in non-production mode in tests, and it uses `eval`.

Switching to `mode: "production"` for tests works.